src/browser/api/window.js

```diff
@@ -1558,24 +1586,81 @@ Window.removeEventListener = function(identity, type, listener) {
     ofEvents.removeListener(route.window(type, browserWindow.id), listener);
 };
 
+function areNewBoundsWithinConstraints(options, width, height) {
+    const {
+        minWidth,
+        minHeight,
+        maxWidth,
+        maxHeight,
+        aspectRatio
+    } = options;
+
+    if (width === undefined && height === undefined) {
+        return true;
+    }
 
-Window.resizeBy = function(identity, deltaWidth, deltaHeight, anchor) {
+    if (height === undefined) {
+        return (width >= minWidth) && (maxWidth === -1 || width <= maxWidth);
+    }
+
+    if (width === undefined) {
+        return (height >= minHeight) && (maxHeight === -1 || height <= maxHeight);
+    }
+
+    const acceptableWidth = (width >= minWidth) && (maxWidth === -1 || width <= maxWidth);
+    const acceptableHeight = (height >= minHeight) && (maxHeight === -1 || height <= maxHeight);
+    const roundedProposedRatio = Math.round(100 * (width / height)) / 100;
```

Comment:
what's the motivation for rounding here? get a 2 decimal point precision? 
I'm not a big fan of comments, but I think this bit is technical enough to justify including some help for future readers

Reply:
Shoutout to my man `Number.toFixed`. 

Some pros/cons vs this method: https://stackoverflow.com/a/12830454
tl;dr: returns a string, so you probably need to change it back to a number, but it handles rounding better in some cases.

^ this is probably a neckbeard nitpick and either version is probably fine for our purposes.

Reply:
Yeah, 2 decimal point precision. Ran a bunch of tests to see how the `aspectRatio` calculation changes depending on width/height, and it seems that our current manual resizing logic has 2-decimal precision. Is this too precise? Not precise enough? Adding comment.

Reply:
@dhamberlin oooooooo really cool, but looking at some of the comments underneath, seems like `Number.toFixed` also has rounding weirdness sometimes. This one in particular caught my eye, what do you think?:

> There is no advantage of using toFixed instead of Math.round; toFixed leads to quite the same rounding problems (try them with 5.555 and 1.005), but is like 500x (no kidding) slower than Math.round – Pierre May 20 '14 at 16:49

Reply:
Good looking out. I like `toFixed` in that its intent is clear (re: Tomer's comment). But this is one of the places in our code base where speed at this scale is liable to actually matter. Haven't benchmarked, but if `Math.round` is faster, we should probably use that.

Reply:
Follow up cause talking perf without testing squicks me out:

Ran some tests and this Pierre fellow is right. My results were a little less dramatic than 500x, but `toFixed` was decidedly slower (took about 2-3 times as long as `Math.round(100 * n) / 100` for most of my test runs)